### ISSUE TITLE
Configuration through environment variables

### ISSUE THREAD

**Issue by thundering-herd** [honeycombio/refinery#275]
Hej,

**tldr; make all configuration properties available via environment variables to provide the means to dynamically adjust `Refinery`**

we are introducing `Refinery` into our setup and stumbled upon the problem that the environment variables are not evaluated within the configuration file.

For example a configuration like following

```
 18 [PeerManagement]
 19 Type = "redis"
 20 RedisHost =  "refinery-leader-redis-${ENVIRONMENT}.mydomain:6379"
```

will not work. (I don't know if this is the behaviour of `viper`).

Looking at the implementation of `Refinery` we can see that only four environment variables are exposed for usage https://github.com/honeycombio/refinery/blob/a8684973018395e11c56e2559535333038660a5d/config/file_config.go#L125

Taking the 12factor principle into account, I can't think about a design decision why not all properties are exposed through environment variables. It seems to be possible with `viper` by defining a `SetEnvPrefix` and running `AutomaticEnv`.

A use case in our scenario is following:

- we are running different environments (e.g. dev, stage, prod)
- each of them uses a dedicated `Refinery` instance (plus `redis`)
- we want to send the metrics of `Refinery` to our analyse backend **but** into different datasets
--> at this moment it is not possible to do this without baking in the different dataset names

Thanks for any input and maybe I am missing something simple here :)

**Comment by olane** [comment]
I'm running into the same issue - I've ended up creating a wrapper Dockerfile which injects the `config.toml` file with various environment variables that we want configurable (or secret) for certain environments. 

It works, but it's still annoying - because it's baked in at the image level we're going to have to build a separate image for each environment. Being able to just pass environment variables in directly would vastly simplify things.

**Comment by olane** [comment]
Ah - so for my use case I didn't realize that there actually do seem to be env variables for the things I care about right now, it's just that the `MetricsAPIKey` value isn't documented as having an env var alternative on https://docs.honeycomb.io/manage-data-volume/refinery/configuration/ 

I'd still be in favour of having as many values as possible have environment variable alternatives, though.

**Comment by cartermp** [comment]
Since Refinery 2.0, more things were added that you can configure via environment variables. So we're thinking of closing this one out since it seems that most things people need are handled now.
